code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -299,7 +299,11 @@ private void populateExternalLinksForService(Service service, Map<String, Object
             if (volumesFrom == null) {
                 volumesFrom = new ArrayList<Integer>();
             }
-            volumesFrom.addAll(volumesFromInstanceIds);
+            for (Integer volumesFromInstanceId : volumesFromInstanceIds) {
+                if (!volumesFrom.contains(volumesFromInstanceId)) {
```

Comment:
I know :) just wanted to avoid the back and forth conversion when retrieving/putting it to/from data field - there its stored in List format